src/main.cc

```diff
@@ -288,11 +288,205 @@ NAN_METHOD(EnumKeys) {
     RegCloseKey(hCurrentKey);
 }
 
-NAN_MODULE_INIT(Init) {
+NAN_METHOD(CreateKey)
+{
+  auto argCount = info.Length();
+  if (argCount != 2)
+  {
+    Nan::ThrowTypeError("Wrong number of arguments");
+    return;
+  }
+
+  if (!info[0]->IsNumber())
+  {
+    Nan::ThrowTypeError("A number was expected for the first argument, but wasn't received.");
+    return;
+  }
+
+  if (!info[1]->IsString())
```

Comment:
I think there is a mismatch between this requirement and the declaration of `createKey` in `registry.ts`:
```typescript
export function createKey(key: HKEY, subkey?: string | null): boolean {
```

According to this, `subkey` could still be undefined (due to `?`) or `null`. If those values are expected, the checks in here should account for them. If those are not a possibility, then this check must be removed (or moved inside the `if (!info[1]->IsNullOrUndefined())` below) and the signature of the Typescript function should be changed like this:
```typescript
export function createKey(key: HKEY, subkey: string): boolean {
```

Maybe that's why you had the argCount check as you had it before? (`argCount != 2 && argCount != 1` IIRC)

Reply:
you're right, the subkey shouldnt be null or undefined. My bad.